projects/packages/waf/src/class-waf-runtime.php

```diff
@@ -278,17 +278,41 @@ public function block( $action, $rule_id, $reason, $status_code = 403 ) {
 		}
 	}
 
+	/**
+	 * Get the headers for logging purposes.
+	 */
+	public function get_request_headers() {
+		$all_headers     = getallheaders();
+		$exclude_headers = array( 'Authorization', 'Cookie', 'Proxy-Authorization', 'Set-Cookie' );
+
+		foreach ( $exclude_headers as $header ) {
+			unset( $all_headers[ $header ] );
+		}
+
+		return $all_headers;
+	}
+
 	/**
 	 * Write block logs. We won't write to the file if it exceeds 100 mb.
 	 *
 	 * @param string $rule_id Rule id.
 	 * @param string $reason Block reason.
 	 */
 	public function write_blocklog( $rule_id, $reason ) {
-		$log_data              = array();
-		$log_data['rule_id']   = $rule_id;
-		$log_data['reason']    = $reason;
-		$log_data['timestamp'] = gmdate( 'Y-m-d H:i:s' );
+		$log_data                 = array();
+		$log_data['rule_id']      = $rule_id;
+		$log_data['reason']       = $reason;
+		$log_data['timestamp']    = gmdate( 'Y-m-d H:i:s' );
+		$log_data['request_uri']  = isset( $_SERVER['REQUEST_URI'] ) ? unslash( $_SERVER['REQUEST_URI'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$log_data['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? unslash( $_SERVER['HTTP_USER_AGENT'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$log_data['referer']      = isset( $_SERVER['HTTP_REFERER'] ) ? unslash( $_SERVER['HTTP_REFERER'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$log_data['content_type'] = isset( $_SERVER['CONTENT_TYPE'] ) ? unslash( $_SERVER['CONTENT_TYPE'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
```

Comment:
```suggestion
		$log_data['request_uri']  = isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
		$log_data['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
		$log_data['referer']      = isset( $_SERVER['HTTP_REFERER'] ) ? wp_unslash( $_SERVER['HTTP_REFERER'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
		$log_data['content_type'] = isset( $_SERVER['CONTENT_TYPE'] ) ? wp_unslash( $_SERVER['CONTENT_TYPE'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
```

Reply:
As a follow-up to this, I think if we are now checking and unslashing these we can remove or modify the `phpcs:ignore`?

Reply:
I ended up using \unslash instead of wp_unslash because we won't have the wp_ version of the function on bootstrap mode. So I'm leaving the phpcs:ignore as it's expecting the wp_unslash to be called.